src/test/java/org/folio/rest/api/HoldingsStorageTest.java

```diff
@@ -791,6 +791,212 @@ public void holdingsCallNumberDoesNotUpdateItemWithItemLevelCallNumber()
   }
 
+  @Test
+  public void updatingHoldingsUpdatesItemEffectiveCallNumberSuffix()
+      throws MalformedURLException, InterruptedException,
+      ExecutionException, TimeoutException {
+
+    UUID instanceId = UUID.randomUUID();
+
+    instancesClient.create(smallAngryPlanet(instanceId));
+
+    UUID holdingId = UUID.randomUUID();
+
+    JsonObject holding = holdingsClient.create(new HoldingRequestBuilder()
+      .withId(holdingId)
+      .forInstance(instanceId)
+      .withPermanentLocation(mainLibraryLocationId)
+      .withCallNumberSuffix("testCallNumberSuffix")
+      .withTags(new JsonObject().put("tagList", new JsonArray().add(TAG_VALUE)))).getJson();
+
+    JsonObject itemToCreate = new JsonObject();
+
+    itemToCreate.put("holdingsRecordId", holdingId.toString());
+    itemToCreate.put("status", new JsonObject().put("name", "Available"));
+    itemToCreate.put("permanentLoanTypeId", canCirculateLoanTypeID);
+    itemToCreate.put("temporaryLocationId", annexLibraryLocationId.toString());
+    itemToCreate.put("materialTypeId", bookMaterialTypeID.toString());
+
+    Response postFirstItemResponse = create(itemsStorageUrl(""), itemToCreate);
+    Response postSecondItemResponse = create(itemsStorageUrl(""), itemToCreate);
+
+    assertThat(postFirstItemResponse.getStatusCode(), is(HttpURLConnection.HTTP_CREATED));
+    assertThat(postSecondItemResponse.getStatusCode(), is(HttpURLConnection.HTTP_CREATED));
+
+    JsonObject firstItemFromPost = postFirstItemResponse.getJson();
+    JsonObject secondItemFromPost = postSecondItemResponse.getJson();
+
+    String firstItemId = firstItemFromPost.getString("id");
+    String secondItemId = secondItemFromPost.getString("id");
+
+    assertThat(firstItemId, is(notNullValue()));
+    assertThat(secondItemId, is(notNullValue()));
+
+    URL getFirstItemUrl = itemsStorageUrl(String.format("/%s", firstItemId));
+    URL getSecondItemUrl = itemsStorageUrl(String.format("/%s", secondItemId));
+
+    Response getFirstItemResponse = get(getFirstItemUrl);
+    Response getSecondItemResponse = get(getSecondItemUrl);
+
+    JsonObject firstItemFromGet = getFirstItemResponse.getJson();
+    JsonObject secondItemFromGet = getSecondItemResponse.getJson();
+
+    assertThat(firstItemFromGet.getString("id"), is(firstItemId));
+    assertThat(firstItemFromGet.getString("holdingsRecordId"), is(holdingId.toString()));
+    assertThat(firstItemFromGet.getJsonObject("effectiveCallNumberComponents").getString("callNumberSuffix"), is("testCallNumberSuffix"));
+    assertThat(secondItemFromGet.getString("id"), is(secondItemId));
+    assertThat(secondItemFromGet.getString("holdingsRecordId"), is(holdingId.toString()));
+    assertThat(secondItemFromGet.getJsonObject("effectiveCallNumberComponents").getString("callNumberSuffix"), is("testCallNumberSuffix"));
+
+    URL holdingsUrl = holdingsStorageUrl(String.format("/%s", holdingId));
+
+    holding.remove("callNumberSuffix");
```

Comment:
I think the `put` below will overwrite an existing value so this remove could likely be omitted